src/Symfony/Component/HttpClient/Tests/HttpClientTraitTest.php

```diff
@@ -215,4 +215,42 @@ public function testPrepareAuthBasic($arg, $result)
         [, $options] = $this->prepareRequest('POST', 'http://example.com', ['auth_basic' => $arg], HttpClientInterface::OPTIONS_DEFAULTS);
         $this->assertSame('Basic '.$result, $options['headers']['authorization'][0]);
     }
+
+    public function provideFingerprints()
+    {
+        foreach (['md5', 'sha1', 'sha256'] as $algo) {
+            $hash = \hash($algo, $algo);
+            yield [$hash, [$algo => $hash]];
+        }
+
+        yield ['AAAA:BBBB:CCCC:DDDD:EEEE:FFFF:GGGG:HHHH:IIII:JJJJ:KKKK', ['pin-sha256' => ['AAAABBBBCCCCDDDDEEEEFFFFGGGGHHHHIIIIJJJJKKKK']]];
+    }
+
+    /**
+     * @dataProvider provideFingerprints
+     */
+    public function testNormalizePeerFingerprint($fingerprint, $expected)
+    {
+        self::assertSame($expected, $this->normalizePeerFingerprint($fingerprint));
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
```

Comment:
we should use the exception from the component's namespace (same below):
`@expectedException \Symfony\Component\HttpClient\Exception\InvalidArgumentException`

Reply:
Oh nice catch. Fixed!